app/services/CheckoutService.scala

```diff
@@ -121,17 +122,22 @@ class CheckoutService(
     val idMinimalUser = authenticatedUserOpt.map(_.user)
 
     val telephoneNumber = subscriptionData.productData match {
-      case Left(paperData) => NormalisedTelephoneNumber.fromStringAndCountry(personalData.telephoneNumber, paperData.deliveryAddress.country orElse personalData.address.country)
+      case Left(paperData) => NormalisedTelephoneNumber.fromStringAndCountry(personalData.telephoneNumber, personalData.address.country orElse paperData.deliveryRecipient.address.country)
       case _ => NormalisedTelephoneNumber.fromStringAndCountry(personalData.telephoneNumber, personalData.address.country)
     }
 
+    val readerType: ReaderType = subscriptionData.productData match {
+      case Left(paperData) if paperData.deliveryRecipient.isGiftee => Gift
+      case _ => Direct  // No gifting support for Digital Pack yet
+    }
+
     val soldToContact = subscriptionData.productData match {
       case Left(paperData) => Some(SoldToContact(
-        name = personalData, // TODO once we have gifting change this to the Giftee's name
-        address = paperData.deliveryAddress,
-        email = personalData.email, // TODO once we have gifting change this to the Giftee's email address
-        phone = telephoneNumber,
-        deliveryInstructions = paperData.sanitizedDeliveryInstructions
+        name = if (readerType == Gift) paperData.deliveryRecipient else personalData,
+        address = paperData.deliveryRecipient.address,
+        email = if (readerType == Gift) paperData.deliveryRecipient.email else Some(personalData.email),
+        deliveryInstructions = paperData.sanitizedDeliveryInstructions,
+        phone = if (readerType == Direct) telephoneNumber else None
```

Comment:
this is minor so we could fix it in another PR if we need to release this, But maybe it would be more clear with different cases for the different reader types:

we could do: 
```scala
(subscriptionData.productData, readerType) match{
case (Left(paperData), Gift) => [gift stuff]
case (Left(paperData), Direct) => [direct stuff]
case (Left(paperData), Agent) => [agent  stuff] //maybe this case should not exist?
case _ => None
```
we could also do this if in the cases and just matching on the productData:
```scala 
 case  Left(PaperData) if (readerType == Gift) => do stuff
```

Reply:
Interesting, I'll have a think about this as I also don't want to add lots of repeated setters in the [stuff] section.